jetcd-core/src/main/java/io/etcd/jetcd/WatchImpl.java

```diff
@@ -98,6 +99,15 @@ public void close() {
         }
     }
 
+    @Override
+    public void requestProgress() {
+        if (!closed.get()) {
+            synchronized (this.lock) {
```

Comment:
Alternatively we can send a single RequestProgress update at this point (we'd need to add a FutureStub property if we go this route) and we can propagate the response to all watchers.

This'll save the etcd server from getting hit with multiple requests if there are multiple watchers